pkg/login/social/okta_oauth_test.go

```diff
@@ -0,0 +1,91 @@
+  "locale": "en_US",
+  "email": "okta-grafana-test@grafana.com",
+  "preferred_username": "okta-grafana-test@grafana.com",
+  "given_name": "Okta",
+  "family_name": "Grafana",
+  "zoneinfo": "UTC",
+  "updated_at": 1669222099,
+  "email_verified": true
+}`
+
+func TestSocialOkta_UserInfo(t *testing.T) {
+	var boolPointer *bool
+	tests := []struct {
+		name                   string
+		userRawJSON            string
+		autoAssignOrgRole      string
+		settingSkipOrgRoleSync bool
+		roleAttributePath      string
+		want                   *BasicUserInfo
+		wantErr                bool
+	}{
+		{
+			name:              "Basic User info",
+			userRawJSON:       testOktaUserRawJSON,
+			autoAssignOrgRole: "Editor",
+			roleAttributePath: "",
+			want: &BasicUserInfo{
+				Id:             "1",
+				Name:           "monalisa octocat",
+				Email:          "octocat@github.com",
+				Login:          "octocat",
+				Role:           "",
+				IsGrafanaAdmin: boolPointer,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				writer.WriteHeader(http.StatusOK)
+				// return JSON if matches user endpoint
+				if strings.HasSuffix(request.URL.String(), "/user") {
+					writer.Header().Set("Content-Type", "application/json")
+					_, err := writer.Write([]byte(tt.userRawJSON))
+					require.NoError(t, err)
+				} else {
+					writer.WriteHeader(http.StatusNotFound)
+				}
+			}))
+			defer server.Close()
+
+			s := &SocialOkta{
+				SocialBase: newSocialBase("okta", &oauth2.Config{},
+					&OAuthInfo{RoleAttributePath: tt.roleAttributePath}, tt.autoAssignOrgRole, false, *featuremgmt.WithFeatures()),
+				apiUrl:          server.URL + "/user",
+				skipOrgRoleSync: tt.settingSkipOrgRoleSync,
+			}
+
+			// create a oauth2 token with a id_token
+			token := &oauth2.Token{}
+			token = token.WithExtra(map[string]interface{}{"id_token": "eyJraWQiOiJLT1FJT2wwbEhtYUJiam1zVVBOald2Y082b1E1eWdvdUhibUtPRms3YUNFIiwiYWxnIjoiUlMyNTYifQ.eyJzdWIiOiIwMHU3ZGZ5Y2kzM2RhYkFjMjVkNyIsIm5hbWUiOiJPa3RhIEdyYWZhbmEiLCJlbWFpbCI6Im9rdGEtZ3JhZmFuYS10ZXN0QGdyYWZhbmEuY29tIiwidmVyIjoxLCJpc3MiOiJodHRwczovL2Rldi0yNTQxNjU0NS5va3RhLmNvbSIsImF1ZCI6IjBvYTdkZnkzb2d4WlZWUE1qNWQ3IiwiaWF0IjoxNjc1MDgyNjUwLCJleHAiOjE2NzUwODYyNTAsImp0aSI6IklELnNvRGlTUkFJR0ZaamI2ZHhCbzUzX3BHc2llaUlqdVFQX3hpVkdySUdUNGciLCJhbXIiOlsicHdkIl0sImlkcCI6IjAwbzZpa3pxempsOXhJUjgwNWQ3IiwicHJlZmVycmVkX3VzZXJuYW1lIjoib2t0YS1ncmFmYW5hLXRlc3RAZ3JhZmFuYS5jb20iLCJhdXRoX3RpbWUiOjE2NzUwNzg5NjUsImF0X2hhc2giOiJKUURfbzhDcGM3U3lBWGVYR1ZvMmZ3In0.Ee9bUOi1Fki4gfuymRhInt_6vzydConbfiemg552U-YlAgYTOGPBrfIoCxSZjBlF0EkONMs3bmmNzfi1CT_McBlEpXnl3KZcHjP2hSKrMLNJX8aUf3FXsL2_tbnjSr_0kuI2r1OrlWjdp7FZ4Inn564Ppq583MaE-qp0EgFpmj7_2bdh_oG8VjDYWM0nqLzEroCYr8rh2yptMBu89MOFnPwzS7ilQdT7zg6FtTgajJb46hVAzR_PeWEJIg1-xZfV4SNjlZ2zfW3d04vnI1k_UE2wTX3L76ew0tUBOscoode_zAvDAD0nPDykLivp8byQkZdfN3wS0k5axP6yTTaBLg"})
```

Comment:
also; is there a way to generate this token from the above configuration? 🤔 

I took this from the api endpoint response value of the `id_token`for when I call the api

Reply:
I think you could try to follow the testing pattern from here: https://github.com/grafana/grafana/blob/main/pkg/login/social/generic_oauth_test.go#L260-L263. Tokens can be encoded using https://jwt.io/ or one of the other tools out there.

 It would allow specifying token for each test case.

Reply:
alright. So i take a arbitrary encoding and hash to encrypt the `id_token` with a specific payload. Il see if I can get that through the test. I tested some tokens from their but could not get the 3 parts and encoding correctly. Let me try some more 👍

Reply:
> Tokens can be encoded using https://jwt.io/ or one of the other tools out there.

+1 , use placeholder tokens instead of real tokens